docs/walkthrough.md

```diff
@@ -0,0 +1,551 @@
+Don't worry too much about what these do for now; the concepts behind them will be explained in depth over the course of the walkthrough.
+
+First, tell us the name of your site. We depend on this in one of our [template functions](template-functions.md):
+
+```python
+SITE_NAME = 'a tiny project'
+```
+
+And your domain; it is used by template functions to turn relative /urls/ into http://actual.absolute/urls/:
+
+```python
+SITE_DOMAIN = 'example.com'
+```
+
+We don't want things set to be offline to appear in the front-end (our pages have on/offline controls), but we definitely want them to appear in our admin:
+
+```python
+PUBLICATION_MIDDLEWARE_EXCLUDE_URLS = (
+    '^admin/.*',
+)
+```
+
+Add our core CMS things to your `INSTALLED_APPS`:
+
+```python
+INSTALLED_APPS = [
+  # .....
+  'cms',
+  'cms.apps.pages',
+  'cms.apps.media',
+  # Links is optional, but it's very handy to have.
+  'cms.apps.links',
+]
+```
+
+Add our context processors to our template context processors (in `['OPTIONS']['context_processors']`):
+
+```python
+'cms.apps.pages.context_processors.pages',
+```
+
+Add the CMS middleware to your `MIDDLEWARE`:
+
+```python
+MIDDLEWARE = [
+  # ...
+  'cms.middleware.PublicationMiddleware',
+  'cms.apps.pages.middleware.PageMiddleware',
+]
+```
+
+And finally, options for our HTML editor:
+
+```python
+WYSIWYG_OPTIONS = {
+    'height': 500,
+    'plugins': [
+        'advlist autolink link image lists charmap hr anchor pagebreak',
+        'wordcount visualblocks visualchars code fullscreen cmsimage hr',
+    ],
+    'toolbar1': 'code | cut copy pastetext | undo redo | bullist numlist | link unlink anchor cmsimage | blockquote',
+    'menubar': False,
+    'toolbar_items_size': 'small',
+    'block_formats': 'Paragraph=p;Header 2=h2;Header 3=h3;Header 4=h4;Header 5=h5;Header 6=h6;',
+    'convert_urls': False,
+    'paste_as_text': True,
+    'image_advtab': True,
+}
+```
+
+## Let's make a content model
+
+First, create an app called "content". Assuming that your apps all live in a folder called `apps`, this will do:
+
+```
+mkdir apps/content
+touch apps/content/__init__.py
+```
+
+Now add your `content` app to your `INSTALLED_APPS`.
+
+Then, add this to your `models.py`:
+
+```python
+from cms.apps.pages.models import ContentBase
+from django.db import models
+
+
+class MyContent(ContentBase):
+    introduction = models.TextField(
+        null=True,
+        blank=True,
+    )
+```
+
+Let's unpack this!
+
+The `Page` model in the CMS defines its place in the hierarchy, some invisible metadata fields, and some publication controls, and that is it.
+It has no opinions about what the _user-visible_ content should look like - not so much as an HTML field!
+This is the job of **content models**.
+
+That's what derivatives of `ContentBase` are. Anything that inherits from `ContentBase` will be available as a page type in the CMS.
+_There is no explicit registration of content models_; it just works.
```

Comment:
I would consider reversing this - first say the `ContentBase` is a derivative of `Page` then talk about what that means.